### PR TITLE
Number of points allocated by y_slices is now set by SpatialMethod

### LINKED ISSUE #179: Move y_slices into SpatialMethod
**Summary** 
Just move y_slices into SpatialMethod because we will need a different number of variables depending on the type of spatialmethod

### PR DESCRIPTION
**Pull request by Scottmar93**
Made the minimal changes so that y_slices now allocates the number of points from the spatial method class.  Added the relevent method to FIniteVolume and SpatialMethodForTesting 

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #179 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Scottmar93** [comment]
Yeah that is true. Implementing this slightly differently now to avoid this

**Comment by Scottmar93** [comment]
Okay removed the function and added self.npts_for_broadcasting to the spatial method so that don't need to redefine the same thing.

File: pybamm/spatial_methods/spatial_method.py [hunk 1]
```diff
@@ -40,6 +40,19 @@ def spatial_variable(self, symbol):
...
+        variable on domain is converted into.
+
+        Parameters
+        ----------
+        domain : str
+            The domain from which we need to get the points
+        """
+
+        return NotImplementedError
+
```
Comment (tinosulzer) [inline pybamm/spatial_methods/spatial_method.py]: should this still be here?

File: tests/shared.py [hunk 2]
```diff
@@ -13,10 +13,18 @@ def __init__(self, mesh):
...
+            dom: submesh.npts for dom, submesh in self.mesh.items()
+        }
+
     def spatial_variable(self, symbol):
         symbol_mesh = self._mesh.combine_submeshes(*symbol.domain)
         return pybamm.Vector(symbol_mesh.nodes)
 
+    def get_num_of_vars(self, domain):
+        return self.mesh[domain].npts
+
```
Comment (tinosulzer) [inline tests/shared.py]: should this still be here?

**Comment by Scottmar93** [comment]
Ah okay i'll merge and sort conflicts

**Comment by Scottmar93** [comment]
Now using mesh[dom].npts_for_broadcast

**Comment by tinosulzer** [comment]
great, have removed unused functions, merging now

**Pull request merged**